client/my-sites/plans/current-plan/thank-you-header/index.js

```diff
@@ -0,0 +1,90 @@
+/** @format */
+/**
+ * External dependencies
+ */
+import PropTypes from 'prop-types';
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import { get, isFinite, noop } from 'lodash';
+import { localize } from 'i18n-calypso';
+
+/**
+ * Internal dependencies
+ */
+import Card from 'components/card';
+import getCurrentPlanPurchaseId from 'state/selectors/get-current-plan-purchase-id';
+import ProgressBar from 'components/progress-bar';
+import QuerySitePurchases from 'components/data/query-site-purchases';
+import { getByPurchaseId } from 'state/purchases/selectors';
+import { getPlan } from 'lib/plans';
+import { getSelectedSiteId } from 'state/ui/selectors';
+import { getSitePlanSlug } from 'state/sites/selectors';
+
+export class PlanSetupHeader extends Component {
+	static propTypes = {
+		progressComplete: PropTypes.number,
+		progressTotal: PropTypes.number,
+	};
+
+	render() {
+		const { moment, purchaseExpiryDate, planName, siteId, translate } = this.props;
+		const duration = purchaseExpiryDate
+			? moment.duration( moment().diff( purchaseExpiryDate ) ).humanize()
+			: null;
+		return (
+			<Card className="thank-you-header">
+				<QuerySitePurchases siteId={ siteId } />
+				<img
+					className="thank-you-header__illustration"
+					alt=""
+					aria-hidden="true"
+					src="/calypso/images/illustrations/fireworks.svg"
+				/>
+				<h1 className="thank-you-header__title">{ translate( 'Thank you for your purchase!' ) }</h1>
+				<p>
+					{ duration && planName
+						? translate(
+								'Your website is on a %(planName)s plan for %(duration)s. Let’s walk through a short checklist of essential security features for safeguarding your website.',
+								{
+									args: { duration, planName },
+								}
+						  )
+						: ' ' /* &nbsp; maintain some space */ }
+				</p>
+				<p>
+					{ translate(
+						'We’ve taken the liberty of starting the first two items, since they’re key to your site’s safety: we’re configuring spam filtering and backups for you now. Once that’s done, we can work through the rest of the checklist.'
+					) }
+				</p>
+				{ /* can be 0 */ isFinite( this.props.progressComplete ) &&
+					/* shouldn't be 0 */ this.props.progressTotal && (
+						<ProgressBar
+							isPulsing
+							total={ this.props.progressTotal }
+							value={ this.props.progressComplete }
+						/>
+					) }
+				<p>
+					<a
+						className="thank-you-header__skip-setup"
+						href={ /* @TODO (sirreal) fix this */ document.location.pathname }
+					>
+						{ translate( 'Skip setup. I’ll do this later.' ) }
+					</a>
+				</p>
+			</Card>
+		);
+	}
+}
+
+export default connect( state => {
+	const siteId = getSelectedSiteId( state );
+	const planSlug = getSitePlanSlug( state, siteId );
+	return {
+		planName: get( getPlan( planSlug ), [ 'getTitle' ], noop )(),
```

Comment:
Do we need the square brackets here?

Reply:
Lodash paths can be provided as dot-joined paths or array paths. `"a"` is roughly equivalent to `[ "a" ]`, but one is the dot-joined path and the other is array path in the same way that `"a.b.c"` is roughly equivalent to `[ "a", "b", "c" ]`.

We don't need the square brackets because we don't need to use the array version, but I've chosen to do so.

Reply:
Sorry for my continuous nitpicking, but I'd just like to hear about rationale for using the array version, in other words, why you chose it in favor of the other one.

Reply:
It allows Lodash to short circuit here and not go through what may be expensive (but memoized) "dot-path" parsing to end up at the array that we can provide initially:
https://github.com/lodash/lodash/blob/67389a8c78975d97505fa15aa79bec6397749807/lodash.js#L4475-L4480

Reply:
Sounds good, thanks again!